docs/content/contributing/general.md

```diff
@@ -24,11 +24,21 @@ To run the test suite, you will need to
 
     We do not support running linting, tests, etc on macOS at this time. Please use a linux VM.
 
+??? tip "Setting the Degree of Parallelization for Tests"
+
+    If your machine is capable, you can increase the amount of tests that are run simultaneously by prepending the `make clean all` command with `BATS_PARALLEL_JOBS=X` (i.e. `BATS_PARALLEL_JOBS=X make clean all`). This wil speed up the test procedure. You can also run all tests in serial by setting `BATS_PARALLEL_JOBS=1` this way.
+
+    The default value of `BATS_PARALLEL_JOBS` is 2. Increasing it to `3` requires 6 threads and 6GB of main memory; increasing it to `4` requires 8 threads and at least 8GB of main memory.
```

Comment:
Ran some tests on my system; I have an older i7-4770k with 8 threads / 4 cores. Running with `--jobs 4` was the maximum when it came to CPU utilisation for me. I'd rather set the bar high and avoid unnecessary test failures due to resource problems than setting it low and risking them. I used `htop` to look at CPU/RAM.

Reply:
I used `--jobs 4` (_with and without `--no-parallelize-within-files`_) on my i5-6500 system (_32GB RAM, 16GB in use_) and for `parallel/set1` I only observed near 2GB peak usage increase via system monitor and reaching 100% CPU load.

The only reason I think you'd have experienced higher RAM usage would perhaps be if the image under test hadn't been built with a recent ClamAV layer updated. I have run tests with the last clean image build being a week or so old and observed ClamAV tests were very memory hungry.

Might have been the case for you? That or there's something different with the latest changes I've pushed for this branch :man_shrugging: 

---

So I'm still not sure how high of a bar you want to set that. Before parallel tests were added, I've seen tests with ClamAV under some circumstances gobble up 6GB+ in memory :man_shrugging: 

Do you have a source for each increment of the `BATS_PARALLEL_JOBS` / `--jobs` requiring double the number of threads? It doesn't seem to be mentioned on the [BATS docs](https://bats-core.readthedocs.io/en/v1.8.2/usage.html).

My understanding is it's just a concurrent queue of jobs that'll get scheduled CPU time on available threads, so since N jobs doesn't necessarily use constant CPU, more CPU time can be available for additional jobs to run?

Reply:
Seems as if the resource requirements differ a lot. The ClamAV update could be an issue, but I don't think so since I built it directly before running the tests, IIRC.

I have no source when it comes to the `--jobs` flag needing more threads. We should therefore probably refrain from using exact numbers here and be more abstract. Something like: "Depending on the hardware in your system, you can increase the number of parallel jobs. Make sure you do not overload the system, which can lead to failing tests."